### PR TITLE
Switch CI to circle and coverage to codecov

### PR DESCRIPTION
**Pull request by jperata**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+COPY lib/ ./lib/
+
+COPY *.json ./
+
+RUN npm -version
+
+RUN npm install
+
+CMD node bin/bst-server.js 443 3000
```
Comment (jkelvie) [inline docker/Dockerfile]: The server port should be 5000 not 3000

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+FROM bespoken/nodejs
+
+MAINTAINER John Kelvie
+
+EXPOSE 3000 443
```
Comment (jkelvie) [inline docker/Dockerfile]: Need to expose 5000 and 443

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+machine:
+  node:
+    version: 6.9.2
```
Comment (jkelvie) [inline circle.yml]: We need to run tests for the various node versions for non-server pieces - the client needs to run on 4-7

Comment (jperata) [inline circle.yml]: that's on the dependencies section, not sure is the actual node version here is still needed , but circle don't have a simple multiple node setup.

**Pull request closed without merging**